### DISCUSSION TITLE
Creating a public.users table with `full_name`

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#2175] [Questions] [answered]
## Goal

I'm following the guide to create a [public.users table](https://supabase.io/docs/guides/auth#create-a-publicusers-table). In the public.users table, I need to store a user's full name.

## Issue

Luckily, when using Google OAuth, a user's full name is stored in `auth.users.raw_user_meta_data.full_name`. However, when I try to access this value in the public.user's trigger, I keep getting an error (in the dashboard logs under Authentication > Settings > Logs). It claims that `column \"full_name\" does not exist`.

```
time="2021-07-01T04:48:41Z" level=error msg="500: Database error saving new user" component=api error="failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: column \"full_name\" does not exist (SQLSTATE 42703)" method=GET path=/callback referer="https://accounts.google.com/" remote_addr="172.90.205.213:40718" request_id=this-is-a-uuid
```

Yet if I query the `auth.users` table right afterwards, I can see that `full_name` exists.

```sql
select raw_user_meta_data from auth.users;
```

<img width="323" alt="Screen Shot 2021-06-30 at 10 00 19 PM" src="https://user-images.githubusercontent.com/5659171/124067053-9169b580-d9ee-11eb-8a47-1ce3d2ab106f.png">

## Code

Here's the function/trigger.

```sql
create or replace function public.handle_new_user() 
returns trigger as $$
begin
  insert into public.users (uid, name)
    values (new.id, new.raw_user_meta_data->full_name);

  return new;
end;
$$ language plpgsql security definer;

create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

**Comment by inian** [comment, marked as answer]
Can you try `new.raw_user_meta_data->'full_name'` @Nick-Mazuk?

**Reply by Nick-Mazuk** [reply]
That gives this error.

```
ERROR: function replace(jsonb, unknown, unknown) does not exist (SQLSTATE 42883)"
```

Using double quotes instead produces the original error.

**Reply by Ehesp** [reply]
Try this:

```
new.raw_user_meta_data::json->'full_name'
```
